liblocation/src/main/java/com/mapbox/android/core/location/GoogleLocationEngine.java

```diff
@@ -88,31 +97,39 @@ public boolean isConnected() {
   }
 
   @Override
-  public void onConnected(@Nullable Bundle bundle) {
-    for (LocationEngineListener listener : locationListeners) {
-      listener.onConnected();
-    }
-  }
-
-  @Override
-  public void onConnectionSuspended(int cause) {
-  }
-
-  @Override
-  public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
+  public void requestLastLocation() {
```

Comment:
Why not mimic the same callback system as google? Why would there be a need for multiple last location update listener? This callback would only be used once. No need to maintain a list, upstream API is able to handle multiple requests.